src/domain/agent/agent/agent.cache.service.ts

```diff
@@ -0,0 +1,84 @@
+import { LogContext } from '@common/enums/logging.context';
+import { AgentInfo } from '@core/authentication';
+import {
+  CACHE_MANAGER,
+  Inject,
+  Injectable,
+  LoggerService,
+} from '@nestjs/common';
+import { Cache } from 'cache-manager';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { getConnection } from 'typeorm';
+import { IAgent, ICredential } from '..';
+@Injectable()
+export class AgentCacheService {
+  constructor(
+    @Inject(CACHE_MANAGER)
+    private readonly cacheManager: Cache,
+    @Inject(WINSTON_MODULE_NEST_PROVIDER)
+    private readonly logger: LoggerService
+  ) {}
+
+  public async getAgentInfoFromCache(
+    email: string
+  ): Promise<AgentInfo | undefined> {
+    return await this.cacheManager.get<AgentInfo>(
+      await this.getAgentInfoCacheKey(email)
+    );
+  }
+
+  public async setAgentInfoCache(agentInfo: AgentInfo): Promise<AgentInfo> {
+    return await this.cacheManager.set<AgentInfo>(
+      await this.getAgentInfoCacheKey(agentInfo.email),
+      agentInfo,
+      {
+        ttl: 1,
```

Comment:
Extract this one and the one in `update` to a constant or config.
isn't 1 second too small? I think boosting it a little will bring more performance benefits